spec/tasks/update_violation_status_for_readings_spec.rb

```diff
@@ -0,0 +1,17 @@
+require 'spec_helper'
+
+describe UpdateViolationStatusForReadings do
+  describe ".exec" do
+    it "updates violation status of readings with outdoor temps" do
+      r1 = create(:reading, :day_time, outdoor_temp: nil, temp: 68)
+      r2 = create(:reading, :day_time, outdoor_temp: nil, temp: 68)
+      r1.outdoor_temp = 54
+      r1.temp = r2.temp = 67
+      r1.save
+      r2.save
+      UpdateViolationStatusForReadings.exec
+      expect(Reading.where.not(violation: false).count).to eq 1
```

Comment:
on these you can 

```
expect { 
  UpdateViolationStatusForReadings.exec
}.to change(Reading.where.not(violation: false).count).from(0).to 1
```
http://www.relishapp.com/rspec/rspec-expectations/v/3-4/docs/built-in-matchers/change-matcher